src/Test/TestAIMuriel.java

```diff
@@ -55,37 +60,65 @@ private void cleanupTest()
   public boolean runTest()
   {
     boolean testPassed = true;
-    testPassed &= validate(testBuildMegatank(), "  Build Megatank test failed.");
-    testPassed &= validate(testHuntStall(), "  Hunting stall-test failed.");
-    testPassed &= validate(testClearAttackRoute(), "  Route clearing test failed.");
-    testPassed &= validate(testWalkInLine(), "  Line walking test failed.");
-    testPassed &= validate(testInfWadeThroughTanks(), "  Infantry move priority test failed.");
-    testPassed &= validate(testTankWadeThroughInfs(), "  Tank move priority test failed.");
+    testPassed &= validate(testForAllAIs((ai) -> testBuildMegatank(ai)), "  Build Megatank test failed.");
+    testPassed &= validate(testForAllAIs((ai) -> testHuntStall(ai)), "  Hunting stall-test failed.");
+    testPassed &= validate(testForAllAIs((ai) -> testClearAttackRoute(ai)), "  Route clearing test failed.");
+    testPassed &= validate(testForAllAIs((ai) -> testWalkInLine(ai)), "  Line walking test failed.");
+    testPassed &= validate(testForAllAIs((ai) -> testInfWadeThroughTanks(ai)), "  Infantry move priority test failed.");
+    testPassed &= validate(testTankWadeThroughInfs(Muriel.info), "  Tank move priority test failed.");
+    // TODO: Consider making Wally pass the above
+//    testPassed &= validate(testForAllAIs((ai) -> testProductionClearing(ai)), "  Infantry move priority test failed.");
+    testPassed &= validate(testProductionClearing(WallyAI.info), "  Free up industry test failed.");
     return testPassed;
   }
 
-  /** Confirm that Muriel will build the correct counter, even when there is only one possible counter. */
-  private boolean testBuildMegatank()
+  private boolean testForAllAIs(Function<AIMaker, Boolean> test)
   {
-    setupTest();
+    boolean testPassed = true;
+
+    for( AIMaker ai : ais )
+    {
+      testPassed &= test.apply(ai);
+      if( !testPassed )
+        break;
+    }
+
+    return testPassed;
+  }
+
+  /** Confirm that the AI will build the correct counter, even when there is only one possible counter. */
+  private boolean testBuildMegatank(AIMaker ai)
+  {
+    setupTest(ai);
 
     // Add an enemy to counter.
     addUnit(testMap, testCo2, "Megatank", 1, 1);
 
-    // Put Muriel in control
-    testCo1.setAIController(testAI);
-
     // Give Muriel resources.
```

Comment:
"Muriel" still referenced all over.

Reply:
And in the file name. I did mention this is a WIP, yes?